### PR TITLE
bpo-37502: handle default parameter for buffers argument of pickle.loads

### PR DESCRIPTION
**Pull request by mmohrhard**
https://bugs.python.org/issue37502

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Our records indicate we have not received your CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Review by aeros — APPROVED** [review]
@mmohrhard Thanks for the contribution! For verification of the bug described in https://bugs.python.org/issue37502, I ran ``pickle.loads(pickle.dumps(1, protocol=pickle.HIGHEST_PROTOCOL), buffers=None)`` in the interpreter after installing the latest version of `python:master`:

image

To test this PR, I installed the branch `mmohrhard:buffer-default-in-pickle-loads` and ran the same code as above in the interpreter:

image

As can be seen from the screenshots above, `TypeError` is no longer raised when buffers is set to `None`.

Edit: I would recommend replacing the news blurb to be a bit more descriptive, to specify the exception. 

Currently it is:

> Fix error when buffer argument of pickle.loads is None

I would recommend changing the phrasing to:

> pickle.loads() no longer raises TypeError when the buffers argument is set to None

**Comment by mmohrhard** [comment]
@aeros167 Thanks. I have updated the patch set with your suggest text for the news blurb and squashed the commits.

Please let me know if there is anything else that I should do.

File: Modules/_pickle.c [hunk 1]
```diff
@@ -1653,7 +1653,7 @@ _Unpickler_SetInputEncoding(UnpicklerObject *self,
 static int
 _Unpickler_SetBuffers(UnpicklerObject *self, PyObject *buffers)
 {
-    if (buffers == NULL) {
+    if (buffers == NULL || buffers == Py_None) {
```
Comment (nirs) [inline Modules/_pickle.c:1656]: Propagating down 2 special values meaning "no buffers" is ugly. Can we 
normalize the value when we parse the arguments, so the rest of the code
have exactly one value for "no buffers"?

**Review by pitrou — APPROVED** [review]
Thanks for spotting this. This looks good to me.

**Pull request merged**